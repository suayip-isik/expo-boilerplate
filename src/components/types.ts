import { GestureResponderEvent } from "react-native";

export interface IMyButton {
  onPress: ((event: GestureResponderEvent) => void) | undefined;
}

export interface IHelloText {
  language: string | undefined;
  name: string | undefined;
}
