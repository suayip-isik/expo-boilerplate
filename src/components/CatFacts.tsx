import { Text } from "react-native";
import React, { useEffect, useState } from "react";

const CatFacts: React.FC = () => {
  const [catFactsData, setCatFactsData] = useState<any>();

  useEffect(() => {
    fetch("https://cat-fact.herokuapp.com/facts")
      .then((response) => {
        response.json();
      })
      .then((data) => {
        setCatFactsData(data);
      });
  }, []);

  return catFactsData && catFactsData.length > 0 ? (
    <Text>{catFactsData[0].text}</Text>
  ) : (
    <Text>No Cat Facts!</Text>
  );
};

export { CatFacts };
