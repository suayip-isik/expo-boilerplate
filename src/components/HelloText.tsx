import { Text, View } from "react-native";
import React from "react";
import { IHelloText } from "./types";
import { setWelcomeText } from "../utils";

const HelloText: React.FC<IHelloText> = ({ language, name }) => {
  return (
    <View>
      <Text>
        {setWelcomeText(language)} {name}
      </Text>
    </View>
  );
};

export { HelloText };
