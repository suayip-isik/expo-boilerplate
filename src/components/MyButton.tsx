import { Text, TouchableOpacity } from "react-native";
import React from "react";
import { IMyButton } from "./types";

const MyButton: React.FC<IMyButton> = ({ onPress }) => {
  return (
    <TouchableOpacity testID="MyButton:Click:ClickMe" onPress={onPress}>
      <Text>this MyButton has been created for testing</Text>
    </TouchableOpacity>
  );
};

export { MyButton };
