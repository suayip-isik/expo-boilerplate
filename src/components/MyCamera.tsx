import { StyleSheet, Text, TouchableOpacity, View } from "react-native";
import React, { useEffect, useState } from "react";
import { Camera, CameraType } from "expo-camera";
import { COLOR_RED, COLOR_WHITE } from "../theme";

const MyCamera: React.FC = () => {
  const [type, setType] = useState(CameraType.back);
  const [permission, requestPermission] = Camera.useCameraPermissions();

  useEffect(() => {
    requestPermission();
    console.log(permission);
  }, []);

  const toggleCameraType = () => {
    setType((current) =>
      current === CameraType.back ? CameraType.front : CameraType.back
    );
  };

  if (!permission?.granted) {
    return <Text>Kamera ya izin verilmemi≈ü </Text>;
  }
  return (
    <Camera style={styles.camera} type={type}>
      <View style={styles.buttonContainer}>
        <TouchableOpacity onPress={toggleCameraType}>
          <Text style={styles.buttonText}>Change Camera</Text>
        </TouchableOpacity>
      </View>
    </Camera>
  );
};

export { MyCamera };

const styles = StyleSheet.create({
  camera: {
    flex: 1,
  },
  buttonContainer: {
    position: "absolute",
    bottom: 20,
    alignSelf: "center",
    borderWidth: 3,
    borderColor: COLOR_WHITE,
    width: 100,
    height: 100,
    borderRadius: 50,
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: COLOR_RED,
  },
  buttonText: {
    color: COLOR_WHITE,
    fontSize: 20,
  },
});
