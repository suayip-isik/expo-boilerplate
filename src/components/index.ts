export * from "./MyButton";
export * from "./HelloText";
export * from "./CatFacts";
export * from "./MyCamera";
