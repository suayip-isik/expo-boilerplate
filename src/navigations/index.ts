export * from "./AuthNavigator";
export * from "./MainTabNavigator";
