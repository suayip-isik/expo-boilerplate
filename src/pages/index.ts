export * from "./HomeScreen";
export * from "./LoginScreen";
export * from "./SettingsScreen";
export * from "./CameraScreen";
