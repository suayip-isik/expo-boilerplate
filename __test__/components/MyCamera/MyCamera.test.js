import React from "react";
import { MyCamera } from "../../../src/components";
import { render, act } from "@testing-library/react-native";

describe("PhotoCamera Tests", () => {
  it("renders correctly", () => {
    const wrapper = render(<MyCamera />);
    expect(wrapper).toMatchSnapshot();
  });
});
