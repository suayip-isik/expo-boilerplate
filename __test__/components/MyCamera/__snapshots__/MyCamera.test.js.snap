// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoCamera Tests renders correctly 1`] = `
<Text>
  Kamera ya izin verilmemi≈ü 
</Text>
`;
